scripts/SasNetworkIntent.psm1

```
@@ -0,0 +1,352 @@
+        }
+        $guestLabel = [string]$return.label
+        Write-Host "NETWORK AUTO-SWITCH: WAB -> saved Guest/Internet [$guestLabel]" -ForegroundColor Cyan
+        try {
+            [void](Invoke-SasSavedWlanConnect -RepoRoot $RepoRoot -ProfileName $guestLabel)
+            $after = Get-SasNetworkIntentState -RepoRoot $RepoRoot
+            if ([string]$after.classification -ne 'GUEST_INTERNET' -or [string]$after.label -ne $guestLabel) {
+                throw "Guest/Internet transition was not proven. Observed: $($after.classification) [$($after.label)] / $($after.authority)"
+            }
+        }
+        catch {
+            try { [void](Invoke-SasSavedWlanConnect -RepoRoot $RepoRoot -ProfileName ([string]$before.network_label)) } catch { }
```

PSScriptAnalyzer (1.25.0)

[warning] Empty catch block is used. Please use Write-Error or throw statements in catch blocks.

(PSAvoidUsingEmptyCatchBlock)